src/Client/Adapter/Curl.php

```diff
@@ -209,6 +209,13 @@ public function connect($host, $port = 80, $secure = false)
             }
         }
 
+        if (isset($this->config['sslcafile']) && $this->config['sslcafile']) {
```

Comment:
That's exactly what I meant.
Instead of throwing an exception (like you suggested), I'd avoid setting `CURLOPT_CAINFO` and `CURLOPT_CAPATH` if `CURLOPT_SSL_VERIFYPEER` is false